### PR TITLE
Backend for vocabulary search

### PR DESCRIPTION
**Pull request by bethanyaconnor**
For vocabulary, we want to search by word and definitions. This use is very similar to what we did with resources and `VocabularyAutocomplete` is pretty similar to `ResourceAutocomplete`. As with resource autocomplete, the test data needs to be in fixtures so that is pre-loaded into the database. We already have course version fixtures so I just reused those for vocab.

If you pull this branch, a URL like http://localhost-studio.code.org:3000/vocabularysearch?limit=7&query=pro&courseVersionId=50 (courseVersionId may need to be changed) should return some results. 

## Testing story

Added unit tests, some manual testing

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by bethanyaconnor** [comment]
Closing for now while I rethink this approach (see [slack discussion](https://codedotorg.slack.com/archives/CNZP84FJ5/p1610592086000800))

**Comment by bethanyaconnor** [comment]
After our discussion at standup, I think this is still a reasonable approach.

File: dashboard/lib/vocabulary_autocomplete.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class VocabularyAutocomplete < AutocompleteHelper
+  def self.get_search_matches(query, limit, course_version_id)
+    limit = format_limit(limit)
+
+    rows = Vocabulary.limit(limit)
+    rows = rows.where(course_version_id: course_version_id)
+    return [] if query.length < MIN_WORD_LENGTH
+    query = format_query(query)
+    rows = rows.
+      where("MATCH(word,definition) AGAINST(? in BOOLEAN MODE)", query)
```
Comment (davidsbailey) [inline dashboard/lib/vocabulary_autocomplete.rb:10]: I totally missed all the work that this line is doing when I reviewed the resources search PR: https://dev.mysql.com/doc/refman/8.0/en/fulltext-search.html Nice find on mysql's fulltext search! 

I am just wondering if you tried the "natural language" mode and had anything to share about why you picked boolean mode? no worries if not, but could be good to capture if so.

**Review by davidsbailey — APPROVED** [review]

File: dashboard/lib/vocabulary_autocomplete.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/lib/vocabulary_autocomplete.rb:10]: I didn't -- for school search (what this logic was initially used for) natural language doesn't make much sense as it was just looking for a school or city name match. I just tried it out and it looks like the results might be better sorted but we lose searching by partial words (eg searching "algo" and "algorithm" popping up).

Comment (davidsbailey) [reply dashboard/lib/vocabulary_autocomplete.rb:10]: Makes sense. Let's see how it goes, and we can always let the curriculum team try out the other way later if there are problems.

**Review by dmcavoy — APPROVED** [review]
Exciting stuff. Vocab is getting so close!

File: dashboard/lib/vocabulary_autocomplete.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/lib/vocabulary_autocomplete.rb:10]: Sounds good!

**Pull request merged**